### PR TITLE
Core/Unit: Fix snare/daze immunities versus SPELL_AURA_USE_NORMAL_MOVEMENT_SPEED.

### PR DESCRIPTION
**Pull request by xjose93**

### CODE REVIEW

**Comment by blindspell** [comment]
after using this patch mounts cant be used

**Comment by xjose93** [comment]
I tested with mounts and also with crusader aura and i didnt find any problem :/. What do you mean?

**Comment by blindspell** [comment]
well after pulling you commit i got reports that people cant use mounts it says interrupted and no mounts could be used then i decided to remove the commit and it worked

**Comment by xjose93** [comment]
I think about interrupted thing is not related to this but i have no idea, anyways try to give more details then i can reproduce and try to fix.

**Comment by blindspell** [comment]
well i have added this to the 4.3.4 branch and the lines are the same so i just pulled the commit and it bugged on it since the bug is the same on the 4.3.4 i thought it would be compatible but it bugs mounts there

**Comment by xjose93** [comment]
In 3.3.5 its ok.

**Comment by funjoker** [comment]
TC for pserver? shame on you -_-'

**Comment by xjose93** [comment]
@funjoker ?

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -12307,10 +12307,21 @@ void Unit::UpdateSpeed(UnitMoveType mtype, bool forced)
...
-                // Use speed from aura
-                float max_speed = normalization / (IsControlledByPlayer() ? playerBaseMoveSpeed[mtype] : baseMoveSpeed[mtype]);
-                if (speed > max_speed)
-                    speed = max_speed;
+                bool shouldImmune = false;
+                if (Creature* creature = ToCreature())
+                {
+                    uint32 immuneMask = creature->GetCreatureTemplate()->MechanicImmuneMask;
+                    if (immuneMask & (1 << MECHANIC_SNARE) || immuneMask & (1 << MECHANIC_DAZE))
+                        shouldImmune = true;
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: can this shouldImmune logic be just a simple break; at this line ?

**Comment by jackpoz** [comment]
could you provide a few creatures with the immuty cases (none, both, only snare, only daze) and a few spells to test ingame the behavior ?

**Comment by xjose93** [comment]
Sindragosa was the one which i tested.

**Comment by jackpoz** [comment]
Tested and debugged with .cast 20184 on Grand Magus Telestra , without the patch the run speed is set to 1, with the patch the run speed is set to 1.4 (and actually not modified since that's the default one)

**Comment by jackpoz** [comment]
thank you for the bug fix!

**Pull request merged**